Newsletters: add 159 (2021-07-28) (#612)
bitcoinops/bitcoinops.github.io: _posts/en/newsletters/2021-07-28-newsletter.md

---
title: 'Bitcoin Optech Newsletter #159'
permalink: /en/newsletters/2021/07/28/
name: 2021-07-28-newsletter
slug: 2021-07-28-newsletter
type: newsletter
layout: newsletter
lang: en
---
This week's newsletter includes our regular sections with the best
questions and answers of the past month from the Bitcoin Stack Exchange,
our latest column about preparing for taproot, a list of new software
releases and release candidates, and descriptions of notable changes to
popular Bitcoin infrastructure software.

## News

*No significant news this week.*

## Selected Q&A from Bitcoin Stack Exchange

*[Bitcoin Stack Exchange][bitcoin.se] is one of the first places Optech
contributors look for answers to their questions---or when we have a
few spare moments to help curious or confused users.  In
this monthly feature, we highlight some of the top-voted questions and
answers posted since our last update.*

{% comment %}<!-- https://bitcoin.stackexchange.com/search?tab=votes&q=created%3a1m..%20is%3aanswer -->{% endcomment %}

> {% assign bse = "https://bitcoin.stackexchange.com/a/" %}

Reviewer: Would it perhaps make sense to prescribe this alias site-wide?

Author: I think it's @bitschmidty's call, since he's using this.  It would change the variable name from `{{bse}}` to `{{site.bse}}` (or `site.whatever_mike_wants`), which might be a tiny bit more effort to type.

- [What is this unusual transaction in the Bitcoin blockchain?]({{bse}}107603)
  Murch describes an output labeled "UNKNOWN" in a [block explorer][topic
  block explorers]. The output is a segwit version 1 output with a contrived

>   pubkey. As pointed out 0xb10c, the 2019 transaction creating this output was for

Reviewer: ```suggestion
  pubkey. As pointed out by 0xb10c, the 2019 transaction creating this output was for
```

>   pubkey. As pointed out 0xb10c, the 2019 transaction creating this output was for
>   purposes of testing segwit v1 support for [Optech's Compatibility

Reviewer: ```suggestion
  pubkey. As pointed out 0xb10c, the 2019 transaction creating this output was for
  the purpose of testing segwit v1 support for [Optech's Compatibility
```

>   Matrix][compat matrix]. As warned previously (see [Newsletter #158][news158
>   taproot]), 0xb10c also inspects some [anyone-can-spend P2TR outputs][0xB10C blog].

Reviewer: This sounds as if we warned that 0xb10c would inspect P2TR outputs, but we warned that anyone could spend them before the activation of taproot.

```suggestion
  Matrix][compat matrix]. As warned previously (see [Newsletter #158][news158
  taproot]), P2TR outputs are anyone-can-spend before the activation of taproot as 0xb10c demonstrated and [elaborates in a blog post][0xB10C blog].
```

- [What are miners signalling for when the block header nversion field ends in 4 i.e. 0x3fffe004?]({{bse}}107443)
  While researching the overt form of ASICBoost, user shikaridota wonders why

>   recently mined blocks have bit 2 being set in the `nVersion` field. Andrew
>   Chow points out that for [taproot][topic taproot], [BIP341's
>   deployment][bip341 deployment] section specifies bit 2 for activation.

Reviewer: ```suggestion
  recently mined blocks have bit 2 being set in the `nVersion` field. Andrew
  Chow points out that [taproot][topic taproot] used bit 2 to signal for activation as specified in [BIP341's
  deployment][bip341 deployment] section.
```

  <!-- [1]: See "Service-Level Based Lifetime Enforcement" in
  https://lightning.engineering/posts/2020-11-02-pool-deep-dive/ -->

   <table>
    <tr>
     <th></th>
     <th>Lightning Pool</th>
     <th>Liquidity advertisements</th>
    </tr>

    <tr>
     <th>Control over listings</th>
     <td>Centralized: allows curation to ensure high-quality listings, but
         also allows censoring listings</td>
     <td>Decentralized: listings can’t be censored, but users need to do
         their own research before buying</td>
    </tr>

    <tr>
     <th>Licensing</th>
     <td>Open source client, proprietary server, open algorithm</td>
     <td>All open source</td>
    </tr>

    <tr>
     <th>Price information</th>
     <td>Actual prices paid are public information via public auction results</td>
     <td>Advertised prices are public information via public gossip network</td>
    </tr>

    <tr>
     <th markdown="1">Third-party purchased liquidity ("[sidecar channels][]")</th>
     <td>Yes, Alice can pay Bob to fund a channel to Carol</td>
     <td>Maybe.  Not specified or implemented, but seems possible</td>
    </tr>

   </table>

> - [BIPs #1072][] merges the informational BIP48 titled "Multi-Script Hierarchy

Reviewer: Let's link to the BIP here.

```suggestion
- [BIPs #1072][] merges the informational [BIP48][] titled "Multi-Script Hierarchy
```

Reviewer: Done, thanks for the suggestion.

  higher than the current rate at which a Bitcoin Core node announces
  addresses.

- [C-Lightning #4669][] fixes several bugs in its [LN offers][topic offers]

>   parsing and validation logic reported by Nadav Igvi.

Reviewer 2: We generally don't attribute individuals in the "notable code changes" section, since the only fair thing to do would be to attribute all authors, reviewers, testers, etc.

```suggestion
  parsing and validation logic.
```
